R/assocTestSingle.R

```diff
@@ -68,25 +79,27 @@ setMethod("assocTestSingle",
                       freq <- freq[keep,,drop=FALSE]
                   }
 
-                  # mean impute missing values
+                  ## mean impute missing values
                   if (any(n.obs < nrow(geno))) {
                       geno <- .meanImpute(geno, freq$freq)
```

Comment:
Need to adjust mean imputation when using recessive or dominant coding.

Reply:
Does it make more sense to impute the frequency before applying the recessive/dominant coding? Something like:
* f < 0.25: round to 0
* 0.25 < f < 0.75: code as 1 for recessive coding, code as 0 for dominant coding
* f > 0.75: code as 1 for recessive and dominant coding

Would probably need to think about the thresholds above, I'm not sure if they're appropriate. Does this even lead to different coding?